app/src/bcsc-theme/contexts/BCSCApiClientContext.tsx

```diff
@@ -35,36 +36,84 @@ export const BCSCApiClientProvider: React.FC<{ children: React.ReactNode }> = ({
-    if (!store.stateLoaded || !store.developer.environment.iasApiBaseUrl) {
+    if (!store.stateLoaded || !store.developer.iasApiBaseUrl) {
       return
     }
 
     const configureClient = async () => {
-      setIsClientReady(false)
-      setError(null)
-
       let newClient = BCSC_API_CLIENT_SINGLETON
 
       try {
         // If the singleton doesn't exist or the base URL has changed, create a new instance
-        if (
-          !BCSC_API_CLIENT_SINGLETON ||
-          BCSC_API_CLIENT_SINGLETON.baseURL !== store.developer.environment.iasApiBaseUrl
-        ) {
-          newClient = new BCSCApiClient(store.developer.environment.iasApiBaseUrl, logger as RemoteLogger)
+        if (!BCSC_API_CLIENT_SINGLETON || BCSC_API_CLIENT_SINGLETON.baseURL !== store.developer.iasApiBaseUrl) {
+          // Only set isClientReady to false when we're actually changing clients
+          setIsClientReady(false)
+          setError(null)
+
+          // Clear old tokens and authentication state when switching environments
+          if (BCSC_API_CLIENT_SINGLETON) {
+            BCSC_API_CLIENT_SINGLETON.tokens = undefined
+            BCSC_API_CLIENT_SINGLETON.tokensPromise = null
+
+            // Check if existing account is from a different environment
+            try {
+              const existingAccount = await getAccount()
+              if (existingAccount) {
+                logger.info('Checking existing account issuer', {
+                  existingIssuer: existingAccount.issuer,
+                  newIssuer: store.developer.iasApiBaseUrl + '/device/',
+                })
+
+                // If the issuer doesn't match the new environment, remove the old account
+                const newIssuer = `${store.developer.iasApiBaseUrl}/device/`
+                if (existingAccount.issuer !== newIssuer) {
+                  logger.info('Account from different environment detected, removing old account', {
+                    oldIssuer: existingAccount.issuer,
+                    newIssuer: newIssuer,
+                  })
+                  await removeAccount()
+                  logger.info('Old account removed successfully')
+                  // Remove metadata as well
+                  dispatch({ type: BCDispatchAction.CLEAR_BCSC })
+                }
+              }
+            } catch (error) {
+              // Rethrow network errors to be handled by outer catch block
+              if (isNetworkError(error)) {
+                throw error
+              }
+              logger.error('Error checking/removing old account', { error })
+              // Continue even if account removal fails
+            }
+
+            // Clear refresh token from store and force re-authentication
+            dispatch({ type: BCDispatchAction.UPDATE_REFRESH_TOKEN, payload: [undefined] })
+            dispatch({ type: BCDispatchAction.UPDATE_VERIFIED, payload: [false] })
+          }
```

Comment:
suggestion: This doesn't feel like the right place for this logic. 

I think we should keep this provider as slim as possible, with it's focus on initializing the client and setting the singleton.